### PR TITLE
replace RowVector(shape...) constructors/calls with RowVector(uninitialized, shape...)

### PR DESCRIPTION
**Pull request by Sacha0**
This pull request deprecates uninitialized-`RowVector` constructors of the form `RowVector{T}(shape...)` in favor of `RowVector{T}(uninitialized, shape...)` equivalents. For background, please see #24595. Best!

Tangentially, these methods do not appear to see use anywhere. Should we simply deprecate these methods altogether?

### CODE REVIEW

**Comment by Sacha0** [comment]
Absent objections or requests for time, I plan to merge these changes this evening PT or later. Best!

**Comment by Sacha0** [comment]
Rebased. Planning to merge after rerunning CI. Best!

File: base/deprecated.jl [hunk 1]
```diff
@@ -2109,6 +2109,12 @@ end
     @deprecate chol!(x::Number, uplo) chol(x) false
 end
 
+# deprecate RowVector{T}(shape...) constructors to RowVector{T}(uninitialized, shape...) equivalents
+@deprecate RowVector{T}(n::Int) where {T}               RowVector{T}(uninitialized, n)
+@deprecate RowVector{T}(n1::Int, n2::Int) where {T}     RowVector{T}(uninitialized, n1, n2)
+@deprecate RowVector{T}(n::Tuple{Int}) where {T}        RowVector{T}(uninitializd, n)
```
Comment (fredrikekre) [inline base/deprecated.jl]: `uninitializd` -> `uninitialized`
I would have thought you were a master of spelling uninitialized after this series of PR's :wink:

**Review by fredrikekre — APPROVED** [review]

File: base/deprecated.jl (hunk 1 above)
Comment (Sacha0) [reply base/deprecated.jl]: I consistently make either of two mistakes: `unintialized`, or `uninitializd`. Seems I have trouble typing vowels? 😄

Comment (Sacha0) [reply base/deprecated.jl]: Fixed. Thanks! :)

**Comment by Sacha0** [comment]
Thanks all!

**Pull request merged**